File: TsinghuaNet.UWP/Converters/EnumIndexConverter.vb
Public Class EnumIndexConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        If [Enum].IsDefined(value.GetType(), value) Then
            Return value
        Else
            Return 0
        End If
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Return value
    End Function
End Class

File: TsinghuaNet.Uno/TsinghuaNet.Uno.Shared/Converters/EnumIndexConverter.cs
using System;
using Windows.UI.Xaml.Data;

namespace TsinghuaNet.Uno.Converters
{
    public class EnumIndexConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            if (Enum.IsDefined(value.GetType(), value))
                return value;
            else
                return 0;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
            => value;
    }
}
